### PR TITLE
[pro#132] Allow pros to add an embargo to an existing request

### PR DESCRIPTION
**Pull request by lizconlan**
Fixes mysociety/alaveteli-professional#132

ToDo: WDTK overrides the sidebar template so will need tweaking to include this addition

### CODE REVIEW

**Comment by lizconlan** [comment]
May require template overrides in the theme

File: app/views/alaveteli_pro/info_requests/_embargo_form.html.erb [hunk 1]
```diff
@@ -1,45 +1,32 @@
...
+      <%= render partial: "alaveteli_pro/info_requests/embargo_extension_form",
+                 locals: {info_request: info_request} %>
+      <%= button_to _("Publish request"),
+                    alaveteli_pro_embargo_path(info_request.embargo),
+                    method: :delete,
+                    data: {
+                      confirm: _("Are you sure you want to publish " \
+                                 "this request?")
+                    } %>
+    <% else %>
```
Comment (gbp) [inline app/views/alaveteli_pro/info_requests/_embargo_form.html.erb]: This `else` conditional will render the create form on batched requests - but when the form is submitted it raises `PermissionDenied` exception [here](https://github.com/mysociety/alaveteli/pull/4287/files#diff-6b4fa2fa47d11a529cdf2a0b0ff63004R15).

While we could prevent render here using an `elsif` and checking to see if `info_request.info_request_batch_id.nil?` I think a nicer solution would be to move this check into the `:create_embargo` ability

File: app/models/ability.rb [hunk 2]
```diff
@@ -72,6 +72,12 @@ def initialize(user)
         can :access, :alaveteli_pro
       end
 
+      # Creating embargoes
+      can :create_embargo, InfoRequest do |info_request|
+        user && info_request.user.is_pro? && !info_request.embargo &&
+          (user == info_request.user || user.is_pro_admin?)
```
Comment (gbp) [inline app/models/ability.rb]: Could move checking for whether or not `info_request_batch_id.nil?` here. Would remove need to limit the create form remove being outputted and remove the check in the controller action too.

File: app/views/alaveteli_pro/info_requests/_embargo_form.html.erb (hunk 1 above)
Comment (lizconlan) [reply app/views/alaveteli_pro/info_requests/_embargo_form.html.erb]: good catch, cheers!

File: app/models/ability.rb (hunk 2 above)
Comment (lizconlan) [reply app/models/ability.rb]: have done - I _think_ we can do that without breaking a future requirement to update batches as that will be for `can :create_embargo, InfoRequestBatch`. I think.

**Pull request closed without merging**